wallet/src/controller.rs

```diff
@@ -660,35 +660,38 @@ where
 fn create_error_response(e: Error) -> Response<Body> {
 	Response::builder()
 		.status(StatusCode::INTERNAL_SERVER_ERROR)
-		.header("access-control-allow-origin", "*")
+		.header(hyper::header::ACCESS_CONTROL_ALLOW_ORIGIN, "*")
 		.header(
-			"access-control-allow-headers",
+			hyper::header::ACCESS_CONTROL_ALLOW_HEADERS,
 			"Content-Type, Authorization",
 		)
+		.header(hyper::header::CONTENT_TYPE, "application/json")
 		.body(format!("{}", e).into())
 		.unwrap()
 }
 
 fn create_ok_response(json: &str) -> Response<Body> {
 	Response::builder()
 		.status(StatusCode::OK)
-		.header("access-control-allow-origin", "*")
+		.header(hyper::header::ACCESS_CONTROL_ALLOW_ORIGIN, "*")
 		.header(
-			"access-control-allow-headers",
+			hyper::header::ACCESS_CONTROL_ALLOW_HEADERS,
 			"Content-Type, Authorization",
 		)
+		.header(hyper::header::CONTENT_TYPE, "application/json")
 		.body(json.to_string().into())
 		.unwrap()
 }
 
 fn response<T: Into<Body>>(status: StatusCode, text: T) -> Response<Body> {
 	Response::builder()
 		.status(status)
-		.header("access-control-allow-origin", "*")
+		.header(hyper::header::ACCESS_CONTROL_ALLOW_ORIGIN, "*")
 		.header(
-			"access-control-allow-headers",
+			hyper::header::ACCESS_CONTROL_ALLOW_HEADERS,
 			"Content-Type, Authorization",
 		)
+		.header(hyper::header::CONTENT_TYPE, "application/json")
```

Comment:
Looks like you can even do something like:

```
.header(hyper::header::ContentType::json())
```
Implies UTF-8 encoding though.

Source: https://hyper.rs/hyper/0.8.0/hyper/header/struct.ContentType.html

Reply:
I was looking for this and couldn't find it, thanks!